### PR TITLE
skip_changelog(ci): gitlab.com support for version updater

### PR DESCRIPTION
**Pull request by anviar**
Support gitlab.com projects for `version_updater.sh`
@SuperQ as we discussed here #294 
Please, check this code, because I don't know how to test it myself completely.
My main idea was to keep code structure unchanged so much as possible.

### CODE REVIEW

File: .github/scripts/version_updater.sh [hunk 1]
```diff
@@ -67,7 +68,16 @@ if [[ -z "${role}" ]]; then
...
+if [[ "${type}" == "github" ]]
+then
+  version="$(github_api "repos/${source_repo}/releases/latest" | jq '.tag_name' | tr -d '"v')"
+elif [[ "${type}" == "gitlab" ]]
+then
+  version="$(curl --retry 5 --silent --fail "https://gitlab.com/api/v4/projects/${source_repo}/releases"|jq '.[0].tag_name'| tr -d '"v')"
+else
+  echo_red 'Unknown source type. Terminating.'
+  exit 128
+fi
```
Comment (gardar) [inline .github/scripts/version_updater.sh:80]: Not sure if this is ever executed, as you are setting github as the default repo_type

File: .github/scripts/version_updater.sh [hunk 2]
```diff
@@ -67,7 +68,16 @@ if [[ -z "${role}" ]]; then
 fi
 
 # Get latest version.
-version="$(github_api "repos/${source_repo}/releases/latest" | jq '.tag_name' | tr -d '"v')"
+if [[ "${type}" == "github" ]]
+then
+  version="$(github_api "repos/${source_repo}/releases/latest" | jq '.tag_name' | tr -d '"v')"
+elif [[ "${type}" == "gitlab" ]]
+then
+  version="$(curl --retry 5 --silent --fail "https://gitlab.com/api/v4/projects/${source_repo}/releases"|jq '.[0].tag_name'| tr -d '"v')"
```
Comment (gardar) [inline .github/scripts/version_updater.sh:76]: Perhaps move the gitlab api url to a function named gitlab_api() to have it in line with the existing github_api() function.

File: .github/scripts/version_updater.sh (hunk 1 above)
Comment (anviar) [reply .github/scripts/version_updater.sh:80]: You right, but in future something can be changed and that condition can handle such situation.

File: .github/scripts/version_updater.sh (hunk 2 above)
Comment (anviar) [reply .github/scripts/version_updater.sh]: I thought about it. I thought this would be an overcomplication of the code.
Okay, I pushed update )

**Review by gardar — APPROVED** [review]

File: .github/scripts/version_updater.sh [hunk 3]
```diff
@@ -67,7 +75,16 @@ if [[ -z "${role}" ]]; then
...
+if [[ "${type}" == "github" ]]
+then
+  version="$(github_api "repos/${source_repo}/releases/latest" | jq '.tag_name' | tr -d '"v')"
+elif [[ "${type}" == "gitlab" ]]
+then
+  version="$(gitlab_api "projects/${source_repo}/releases" | jq '.[0].tag_name' | tr -d '"v')"
+else
+  echo_red 'Unknown source type. Terminating.'
+  exit 128
+fi
```
Comment (SuperQ) [inline .github/scripts/version_updater.sh:87]: Let's use a case statement here. Looks cleaner than a bunch of ifs.

```suggestion
case "${type}" in
  github)
    version="$(github_api "repos/${source_repo}/releases/latest" | jq '.tag_name' | tr -d '"v')"
    ;;
  gitlab)
    version="$(gitlab_api "projects/${source_repo}/releases" | jq '.[0].tag_name' | tr -d '"v')"
    ;;
  *)
    echo_red 'Unknown source type. Terminating.'
    exit 128
    ;;
esac
```

**Review by SuperQ — APPROVED** [review]

**Pull request merged**